### PR TITLE
Docs updates for Cloud - Jan 22

### PR DESCRIPTION
**Pull request by ryphil**
Changes include: 
- Added link to provide org instructions to provide organizational access on GitHub
- Additional clarity around when public repos are made private
- Additional clarity around metadata usage on cloud
- Various additional iterations based on user feedback

### CODE REVIEW

**Review by unknwon — APPROVED** [review]
👍

File: doc/cloud/index.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 - [Getting Started on Sourcegraph Cloud](getting_started_on_cloud.md)
 - [Differences between Sourcegraph Cloud, self-hosted, and enterprise](cloud_ent_on-prem_comparison.md)
 - [Indexing open source code in Sourcegraph Cloud](indexing_open_source_code.md)
 - [Notifications on Sourcegraph Cloud](notifications_on_cloud.md)
 - [Access tokens on Sourcegraph Cloud](access_tokens_on_cloud.md)
 
-## Sourcegraph Cloud for Organizations
-Note: Sourcegraph Cloud for Organizations is in Private Beta. If you are interested in getting early-access, [join our waitlist](https://share.hsforms.com/14OQ3RoPpQTOXvZlUpgx6-A1n7ku)!
+## Sourcegraph Cloud for Teams
+Sourcegraph Cloud for Teams allows small teams to experience the best of Sourcegraph, without self-hosting. Sourcegraph Cloud for Teams is in Private Beta. Learn more [here](https://about.sourcegraph.com/cloud-beta/) and [join our waitlist](https://share.hsforms.com/14OQ3RoPpQTOXvZlUpgx6-A1n7ku) to get access today!
```
Comment (quinnkeast) [inline doc/cloud/index.md:12]:
```suggestion
Sourcegraph Cloud for Teams helps small teams to search across their code together and increase developer velocity through the easiest way to deploy Sourcegraph. Sourcegraph Cloud for Teams is in Private Beta. Learn more [here](https://about.sourcegraph.com/cloud-beta/) and [join our waitlist](https://share.hsforms.com/14OQ3RoPpQTOXvZlUpgx6-A1n7ku) to get access today!
```

A few subtle tweaks proposed here: "without self-hosting" implies there's a comparison between cloud and self-hosting taking place in this moment in time. As we don't mention self-hosting elsewhere on this page, I don't think we need to be quite so explicit about the alternative (because if we are, then it may work against us by creating a moment of "wait, what's self-hosted then?"). As well, "experience the best of Sourcegraph" makes it more about "using Sourcegraph" than "doing what you want to do with Sourcegraph."

I don't actually like the proposed change exactly, and think it needs another iteration or two, but it conveys what I mean.

File: doc/cloud/organizations/code_visibility_teams_sourcegraph_cloud.md [hunk 2]
```diff
@@ -5,12 +5,14 @@ Sourcegraph Cloud protects your organization’s private code using repository p
 
 ## Public repositories
 
-If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context.
+If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context. If a public repository is moved to private on the code host, Sourcegraph will respect the updated permissions.
```
Comment (quinnkeast) [inline doc/cloud/organizations/code_visibility_teams_sourcegraph_cloud.md:8]:
```suggestion
If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context. If a public repository is made private on the code host, that repository will no longer appear in the global search context. 
```

File: doc/cloud/organizations/code_visibility_teams_sourcegraph_cloud.md [hunk 3]
```diff
@@ -5,12 +5,14 @@ Sourcegraph Cloud protects your organization’s private code using repository p
 
 ## Public repositories
 
-If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context.
+If a repository is public on GitHub.com or GitLab.com, other users on Sourcegraph Cloud can view and search across that repository. The repository will appear in the global search context. If a public repository is moved to private on the code host, Sourcegraph will respect the updated permissions. 
 
 ## Private repositories
 
+TL;DR: Users on Sourcegraph can only see a private repository if they have permissions to see that repository on your code host and if they are a member of the organization on Sourcegraph.
```
Comment (quinnkeast) [inline doc/cloud/organizations/code_visibility_teams_sourcegraph_cloud.md:12]:
```suggestion
TL;DR: Users on Sourcegraph can only see a private repository if they have permissions to see that repository on your code host and if they are a member of your organization on Sourcegraph. 
```
